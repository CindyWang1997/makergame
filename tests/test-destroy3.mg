extern void end_game();

object helper { }

object main {
  event create {
    int i;
    for (i = 0; i < 5; i = i + 1) create helper;
    /* outer loop should not iterate over things that were destroyed */
    foreach (helper x) {
      printstr("outer");
      foreach (helper y) {
        printstr("inner");
        destroy y;
      }
    }
    end_game();
  }
}
