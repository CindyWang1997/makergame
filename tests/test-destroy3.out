outer
inner
inner
inner
inner
inner
